### DISCUSSION TITLE
Wordpress implementation issue for education website

### DISCUSSION THREAD

**Discussion by farouq-11** [google/model-viewer#2883] [Q&A] [answered]
Hello, I am trying to use model viewer on my WordPress website designed to help with convey basic civil engineering principles. I have no coding experience and strictly focus on physics and civil engineering related stuff. 

I have been beating my head for a few days trying to get model viewer to load on certain pages. The quickest and easiest method appears to be to use a plugin called "insert headers and footers" and simply copy and paste the <script type="module" src="https://unpkg.com/@google/model-viewer/dist/model-viewer.min.js"></script> in the header section. 

From there the documentation recommends inserting the html code into a page. 

Well, this almost works, but unfortunately the model does not load. Please see the image below showing how the load bar is stuck without a corresponding model to view. 

![image](https://user-images.githubusercontent.com/92694711/137645835-9a47e40e-1a1a-47f3-b19f-fd2f43585832.png)

When looking at the console details, I am seeing the following error: THREE.RGBELoader Bad File Format: bad initial token

![image](https://user-images.githubusercontent.com/92694711/137645888-3c3f446d-2720-40a8-b37e-8ecaebd1ef5d.png)

Lastly, the script inserted in the header section is as follows: 

![image](https://user-images.githubusercontent.com/92694711/137645970-0f31e1f6-1c64-436b-ba15-e3653714ec16.png)

Any help or insights are greatly appreciated.

**Comment by ThorstenBux** [comment, marked as answer]
I would host modelviewer on a different website and then just load that using an iFrame. 
Have you tried using an iFrame to include the model?

**Reply by farouq-11** [reply]
Hi Thorsten, no I have not. Is there documentation or a post I can follow to use this approach? Thank you

**Reply by ThorstenBux** [reply]
Not that I know of but this should do:
```
<p><iframe src="<<URL here>>"; width="100%" height="500px" frameborder="0" allow="xr-spatial-tracking" allowfullscreen=""></iframe></p>
```

As a test you can use this URL: https://webar.tripod-digital.co.nz/?exp=https://storage.googleapis.com/ar-tee.appspot.com/art/swarez_gallery/swarez_gallery

An example is here: https://tripod-digital.co.nz/art (down on the page)

**Reply by farouq-11** [reply]
Dang it! It won't work. It may be because of the theme I'm using (kadence theme). Are you using an original WordPress theme for your website? 

![image](https://user-images.githubusercontent.com/92694711/137646845-7fe759bb-9e6b-415d-8bb4-93ed10b32400.png)

![image](https://user-images.githubusercontent.com/92694711/137646854-e6d123d6-b65a-4deb-b15a-6cd35eb98a69.png)

I'll see if I can try different things, getting closer!

**Reply by ThorstenBux** [reply]
Are you hosting on a .wordpress domain? wordpress blocks iframe if you are using their domain to host.

**Comment by farouq-11** [comment]
No, not a .wordpress domain. Using google domain - directhub.net

I am sure iframe works on the website since I have used a
https://vectary.com/ iframe embed before. That was just wasn't working for
what I am trying to do. Namely, animations and annotation features.

**Reply by ThorstenBux** [reply]
If you can share your URL where it is deployed I can have a look

**Comment by farouq-11** [comment]
Thank you!  I have created a test page here:

**Reply by ThorstenBux** [reply]
<img width="697" alt="Screen Shot 2021-10-18 at 11 30 14 AM" src="https://user-images.githubusercontent.com/16772796/137647224-d6cc4415-6f07-47cd-81ca-4cffae52a7c4.png">
Your iframe source contains too much stuff. Especially HTML elements. Make sure there is only the URL in there. It will work once that is cleaned up

**Comment by farouq-11** [comment]
Oh god, thank you!!!! Should've known better by removing <<>> "insertion
operation and output operator" - so in programming language that literally
means insertion - had to google that. Thank you so much Thorston, this has
made my day!
